src/bin/wac.rs

```diff
@@ -26,19 +29,103 @@ enum Wac {
 async fn main() -> Result<()> {
     pretty_env_logger::init();
 
-    if let Err(e) = match Wac::parse() {
-        Wac::Parse(cmd) => cmd.exec().await,
-        Wac::Resolve(cmd) => cmd.exec().await,
-        Wac::Encode(cmd) => cmd.exec().await,
-    } {
-        eprintln!(
-            "{error}: {e:?}",
-            error = "error".if_supports_color(Stream::Stderr, |text| {
-                text.style(Style::new().red().bold())
-            })
-        );
-        std::process::exit(1);
-    }
-
+    with_interactive_retry(|retry: Option<Retry>| async {
+        if let Err(e) = match Wac::parse() {
+            Wac::Parse(cmd) => cmd.exec().await,
+            Wac::Resolve(cmd) => cmd.exec(retry).await,
+            Wac::Encode(cmd) => cmd.exec(retry).await,
+        } {
+            match e {
+                CommandError::General(e) => {
+                    eprintln!(
+                        "{error}: {e:?}",
+                        error = "error".if_supports_color(Stream::Stderr, |text| {
+                            text.style(Style::new().red().bold())
+                        })
+                    );
+                    std::process::exit(1);
+                }
+                CommandError::Serde(e) => {
+                    eprintln!(
+                        "{error}: {e:?}",
+                        error = "error".if_supports_color(Stream::Stderr, |text| {
+                            text.style(Style::new().red().bold())
+                        })
+                    );
+                    std::process::exit(1);
+                }
+                CommandError::Wac(e) => {
+                    eprintln!(
+                        "{error}: {e:?}",
+                        error = "error".if_supports_color(Stream::Stderr, |text| {
+                            text.style(Style::new().red().bold())
+                        })
+                    );
```

Comment:
Seems like there's a lot of repeat of this colored error string. Can we clean that up?